jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -1474,52 +1475,56 @@ public void handle(Runnable runnable)
      */
     public boolean isProtectedTarget(String target)
     {
-        if (target == null || _protectedTargets == null)
+        if (target == null || _protectedTargets.isEmpty())
             return false;
 
-        while (target.startsWith("//"))
-        {
+        if (target.startsWith("//"))
```

Comment:
Because we are matching only prefixes from the index.  So firstly we'd never see `/foo//////../foo-inf` here as it would be normalised to `/foo//////foo-inf`, then that will not match as it should not match. Even if the file system ignores the empty segments, it will the same as `/foo/foo-inf` which still shouldn't match.

Ultimately we only need to remove leading multiple //// but removing them all is simpler code and the extra cycles are not an issue as it should be very rare to see a target starting with `//`